optuna/storages/_rdb/alembic/versions/v3.0.0.c.py

```diff
@@ -89,20 +89,30 @@ def upgrade():
 
     session = orm.Session(bind=bind)
     try:
-        records = session.query(IntermediateValueModel).all()
+        records = (
+            session.query(IntermediateValueModel)
+            .filter(
+                sa.or_(
+                    IntermediateValueModel.intermediate_value > 1e16,
+                    IntermediateValueModel.intermediate_value < -1e16,
```

Comment:
`1e16` is a heuristic number to filter the target records.

Please note that I tried using `RDB_MAX_FLOAT - 1` instead of `1e16` here (299d2379bdebe9f77a5a64c7361e90c76cfdb617), but it did not work actually. So I use `1e16` as a heuristic threshold that can filter most of intermediate values.

Reply:
I think in this particular case we could use `>= RDB_MAX_FLOAT`. This would not introduce false positives.

Reply:
I don7t think we could use `>= RDB_MAX_FLOAT` since `> RDB_MAX_FLOAT - 1` didn't work. It may cause the representation error of floating point numbers.

Reply:
`> RDB_MAX_FLOAT-1` doesn't work because `RDB_MAX_FLOAT - 1 == RDB_MAX_FLOAT` in double precision. `>= RDB_MAX_FLOAT` should work.

Reply:
> I think in this particular case we could use `>= RDB_MAX_FLOAT`. This would not introduce false positives.

I  tried this approach, and I could migrate data as expected.

Note that some database including PostgreSQL and MSSQL do not accept numpy number type, so I needed to convert `RDB_MAX_FLOAT` to built-in float type with `RDB_MAX_FLOAT.item()`.

```console
[E 2022-10-04 07:55:35,159] (psycopg2.ProgrammingError) can't adapt type 'numpy.float32'
[SQL: SELECT trial_intermediate_values.trial_intermediate_value_id AS trial_intermediate_values_trial_intermediate_value_id, trial_intermediate_values.intermediate_value AS trial_intermediate_values_intermediate_value, trial_intermediate_values.intermediate_value_type AS trial_intermediate_values_intermediate_value_type 
FROM trial_intermediate_values 
WHERE trial_intermediate_values.intermediate_value >= %(intermediate_value_1)s OR trial_intermediate_values.intermediate_value <= %(intermediate_value_2)s OR trial_intermediate_values.intermediate_value IS NULL]
[parameters: {'intermediate_value_1': 3.4028235e+38, 'intermediate_value_2': -3.4028235e+38}]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

Do you think if any databases make stored values smaller than original values, for example, due to numerical error or limitation of internal representation?

Reply:
Oh, I see. I'm sorry that my opinion seemed wrong. And thank you yanase-san for checking the approach.

I am not sure that this logic (`>= RDB_MAX_FLOAT`) will always work correctly in any database. While there may be cases where `>=RDB_MAX_FLOAT` could be used to further speed up the data migration, this heuristic `1e16` still seems reasonable. It doesn't matter if the value is sufficiently large to reduce the number of records in order to speed up data migration.

What do you think?

Reply:
Thank you for your comment. IMO, the current heuristic thresholds are acceptable since they just filter the record to accelerate the migration. The logic to convert `RDB_MAX_FLOAT` to `inf` is not changed in this PR.